dataprofiler/tests/profilers/profiler_options/test_profiler_class_options.py

```diff
@@ -0,0 +1,116 @@
+    
+        # Treat is_enabled as a BooleanOption
+        for key in option.structured_options.properties:
+            expected_error = "type object 'structured_options.{}.is_enabled' has no attribute 'is_enabled'".format(key)
+            with self.assertRaisesRegex(AttributeError, expected_error):
+                option.set({'structured_options.{}.is_enabled.is_enabled'.format(key): True})
+    
+    def test_validate_helper(self):
+        # Valid cases should return [] while invalid cases should return a list of errors
+        option = self.get_options()
+        optpth = self.get_options_path()
+
+        # Default Configuration Is Valid
+        self.assertEqual([], option._validate_helper())
+        
+        # Variable Path Is Not A String
+        expected_error = "The variable path must be a string."
+        with self.assertRaisesRegex(ValueError, expected_error):
+            option._validate_helper(1)
+        
+        # Option is_enabled is not a boolean
+        for key in option.structured_options.properties:
+            option.set({'structured_options.{}.is_enabled'.format(key): "Hello World"}) 
+
+        expected_error = ['{}.structured_options.{}.is_enabled must be a Boolean.'.format(optpth, key) 
+            for key in option.structured_options.properties]
+        expected_error = set(expected_error)
+        # Verify expected errors are a subset of all errors
+        self.assertSetEqual(expected_error, expected_error.intersection(set(option._validate_helper())))
+
+        # Wrong Class Type
+        option = self.get_options()
+        option.structured_options = ProfilerOptions()
+
+        expected_error = ['{}.structured_options must be a StructuredOptions.'.format(optpth,)]
+        self.assertEqual(expected_error, option._validate_helper())
+            
+    def test_validate(self):
+        # Valid cases should return None while invalid cases should return or throw a list of errors
+        option = self.get_options()
+        optpth = self.get_options_path()
+    
+        # Default Configuration Is Valid
+        self.assertEqual(None, option.validate())
+        
+        # Option is_enabled is not a boolean
+        for key in option.structured_options.properties:
+            option.set({'structured_options.{}.is_enabled'.format(key): "Hello World"}) 
+
+        expected_error = ['{}.structured_options.{}.is_enabled must be a Boolean.'.format(optpth, key) 
+            for key in option.structured_options.properties]
+        expected_error = set(expected_error)
+        # Verify expected errors are a subset of all errors
+        with self.assertRaises(ValueError) as cm:
+            option.validate(raise_error=True)
+        raised_error = set(str(cm.exception).split("\n"))
+        self.assertEqual(expected_error, expected_error.intersection(raised_error))
+        self.assertSetEqual(expected_error, expected_error.intersection(set(option.validate(raise_error=False))))
```

Comment:
This feels unnecessary as this is essentially the validate_helper test and you have a more robust version of this that handles the error being raised in the lines above, I think you can remove this because it makes this block a bit overwhelming.

Reply:
It is a different method though - for the same reason the set and set_helper look very similar, it's just to test that both _validate_helper and validate are both returning the same errors (and if something changes it will be caught here).

Reply:
As long as validate calls _validate_helper correctly though, you are correct in that it is a bit unnecessary

Reply:
But in my mind if they don't return the same thing it'll get caught in the fact that the same test cases and expected output across 2 test functions will have one pass and one not, so it feels unnecessary. This isn't critical or anything, just trying to find places to slim these functions down because they are quite lengthy.

Reply:
That's true, just to clarify are you calling for removing all occurrences of raise_error=False or this entire block?